solutions/cfgym-101102D.mdx

```diff
@@ -0,0 +1,263 @@
+---
+id: cfgym-101102D
+source: CF
+title: Rectangles
+author: Kevin Sheng
+---
+
+Let's first simplify this problem, and then generalize the solution of that problem to this one.
+
+Say we had a matrix of all `0`'s and `1`'s. How many submatrices of all `1`'s are there?
+
+Let's first define a 2D array, $\texttt{longest\_consec}$, that contains the number of consecutive `1`'s that are directly to the right for each cell (including the cell itself).  
+For example, this matrix:
+
+<center>
+<table>
+	<tr>
+		<td>1</td>
+		<td>0</td>
+		<td>1</td>
+		<td>1</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>1</td>
+		<td>1</td>
+		<td>1</td>
+	</tr>
+	<tr>
+		<td>0</td>
+		<td>0</td>
+		<td>0</td>
+		<td>1</td>
+	</tr>
+</table>
+</center>
+
+Would have the following $\texttt{longest\_consec}$:
```

Comment:
```suggestion
would have the following $\texttt{longest\_consec}$:
```
No need for capitalization because it's a continuation of the previous sentence